pygments/lexers/nimrod.py

```diff
@@ -118,6 +133,18 @@ def underscorize(words):
             (r'\$', String)
             # newlines are an error (use "nl" state)
         ],
+        'doccomment': [
+            (r'[^\]#]', String.Doc),
+            (r'##\[', String.Doc, '#push'),
+            (r'\]##', String.Doc, '#pop'),
+            (r'[\]#]', String.Doc),
+        ],
+        'comment': [
+            (r'[^\]#]', Comment.Multiline),
```

Comment:
```suggestion
            (r'[^\]#]+', Comment.Multiline),
```

To avoid one token per char (see CONTRIBUTING.md).